find_centroids.py

```diff
@@ -107,17 +144,18 @@ def ret_centres(image, params) :
     #loop over circles and find bolts, end up with list of circles, each a list of bolts (no circle parameters are saved )
     for i in circles[0, :] :
         bolt_ring = ret_centres(bolts_gray, i)
-        bolt_centres.append(bolt_ring)
-    
+        bolts_ord = order_bolts(i, bolt_ring)
+
         #Draw hough circles on img
         cv2.circle(img,(i[0],i[1]),i[2],(0,255,0),2)
         cv2.circle(img,(i[0],i[1]),2,(0,255,0),3)
 
-
-    #draw bolt locations
-    for i in bolt_centres :
-        for j in i :
-            cv2.circle(img, (j[1], j[0]), 5, (0,0,255), -1)
+        #draw bolt locations and number them 
+        bolt_no = 0
+        for i in bolts_ord :
+            bolt_no += 1
+            cv2.putText(img, f'{bolt_no}', (i[1], i[0]), cv2.FONT_HERSHEY_SIMPLEX, 1, (255,255,255), 1)
```

Comment:
Changed to a different color to make numbers visible against the white bolt masks